src/emqtt.erl

```diff
@@ -1380,3 +1509,72 @@ reason_code_name(16#A0) -> maximum_connect_time;
 reason_code_name(16#A1) -> subscription_identifiers_not_supported;
 reason_code_name(16#A2) -> wildcard_subscriptions_not_supported;
 reason_code_name(_Code) -> unknown_error.
+
+%%--------------------------------------------------------------------
+%% Process Dictionary
+
+%% Added for 1.2.3.4
+%%
+%% Due to compatibility issues, we can't directly modify the structure of the state to store
+%% these newly added configurations. So we put `max_retry_count` and `message_delivery_timeout`
+%% into the process dictionary temporarily.
+put_max_retry_count(N) when is_integer(N) ->
+    erlang:put(max_retry_count, N).
+
+get_max_retry_count() ->
+    Default = 3,
```

Comment:
```suggestion
    Default = 0,
```

Reply:
Some test cases require emqtt to do retransmissions, so changing it to 0 here may break compatibility.